packages/functional-tests/src/runner/test-runner.spec.ts

```diff
@@ -31,97 +32,186 @@ class TestGroupStub {
     }
 }
 
+class TestGroupBStub {
+    @test(TestEnvironment.all)
+    public testE(): void {
+        console.log('Invoked test E');
+    }
+}
+
 describe(TestRunner, () => {
-    let testContainer: TestGroupStub;
-    let testContainerName: string;
+    const runId = 'run id';
+    const releaseId = 'release id';
+    let testContainerA: TestGroupAStub;
+    let testContainerB: TestGroupBStub;
+    let testContainerAName: string;
+    let testContainerBName: string;
     let testRunner: TestRunner;
     let loggerMock: IMock<Logger>;
+    let guidGeneratorMock: IMock<GuidGenerator>;
 
     beforeEach(() => {
-        testContainer = new TestGroupStub();
-        testContainerName = testContainer.constructor.name;
+        testContainerA = new TestGroupAStub();
+        testContainerB = new TestGroupBStub();
+        testContainerAName = testContainerA.constructor.name;
+        testContainerBName = testContainerB.constructor.name;
         loggerMock = Mock.ofType();
-        testRunner = new TestRunner(loggerMock.object);
+
+        guidGeneratorMock = Mock.ofType<GuidGenerator>();
+        guidGeneratorMock.setup(g => g.createGuid()).returns(() => runId);
+
+        testRunner = new TestRunner(loggerMock.object, guidGeneratorMock.object);
     });
 
     afterEach(() => {
         loggerMock.verifyAll();
     });
 
-    it('run tests for the given environment only', async () => {
-        loggerMock.setup(o => o.log(It.isAny(), It.isAny(), It.isAny())).verifiable(Times.exactly(3));
+    it('run all tests', async () => {
+        loggerMock.setup(o => o.trackEvent('FunctionalTest', It.isAny())).verifiable(Times.exactly(5));
         loggerMock
             .setup(o =>
-                o.log(`[E2E] Test ${testContainerName}.testC pass`, LogLevel.info, {
-                    source: 'e2e',
-                    testContainer: testContainerName,
-                    testName: 'testC',
+                o.trackEvent('FunctionalTest', {
+                    runId: runId,
+                    releaseId: releaseId,
                     environment: 'canary',
+                    testContainer: testContainerAName,
+                    testName: 'testC',
                     result: 'pass',
                 }),
             )
             .verifiable(Times.once());
         loggerMock
             .setup(o =>
-                o.log(`[E2E] Test ${testContainerName}.testD pass`, LogLevel.info, {
-                    source: 'e2e',
-                    testContainer: testContainerName,
+                o.trackEvent('FunctionalTest', {
+                    runId: runId,
+                    releaseId: releaseId,
+                    environment: 'canary',
+                    testContainer: testContainerAName,
                     testName: 'testD',
+                    result: 'pass',
+                }),
+            )
+            .verifiable(Times.once());
```

Comment:
can u create a helper method to avoid duplicate code? setupFunctionalTestTelemetry(testContainer, testName, state)?